pkg/cover/backend/elf.go

```diff
@@ -196,3 +176,43 @@ func elfReadModuleCoverPoints(target *targets.Target, module *Module, info *symb
 	}
 	return pcs, nil
 }
+
+func alignUp(addr, align uint64) uint64 {
+	if align == 0 {
+		return addr
+	}
+	return (addr + align - 1) & ^(align - 1)
+}
+
+// Calculate the offset of the module .text section in the kernel memory.
+// /proc/modules only contains the base address, which corresponds to the
+// beginning of the first code section, but that section does not have to
+// be .text (e.g. on Android it may be .plt).
+// The offset is calculated by summing up the aligned sizes of sections
+// that:
+//   - precede .text;
+//   - are not .init sections;
+//   - have the same section flags as .text.
+func elfGetModuleOffset(path string) uint64 {
+	file, err := elf.Open(path)
+	if err != nil {
+		return 0
+	}
+	defer file.Close()
+	ts := file.Section(".text")
+	if ts == nil {
+		return 0
+	}
+	off := uint64(0)
+	textFlags := ts.SectionHeader.Flags
```

Comment:
Potentially we need to look only at AX flags:
https://elixir.bootlin.com/linux/latest/source/kernel/module/main.c#L1514

I see in some modules WAX section:
```
  [24] .text.ftrace[...] PROGBITS         0000000000000342  00002401
       0000000000000001  0000000000000000 WAX       0     0     1
```
Not sure what happens with them. Are they loaded into .text segment? Is the segment writable or not? Perhaps `.text.ftrace` has wrongly set `W` and loaded into non-writable segment.

Reply:
Pixel modules I am looking at all have AX, not WAX, for .text.ftrace

But I agree with you that we only need to look at AX, as the mask is only checking them.